### DISCUSSION TITLE
Creating jar repeatedly for custom rule- any other method ?

### DISCUSSION THREAD

**Discussion by Prapti08** [pmd/pmd#4222] [Q&A] [answered]
For checking custom rules I'm changing the code for that particular rule so every time I have to create the jar again is there any other method or way to do this?
It's very time-consuming to create a jar again and again.
Please help me. Thanks in advance.

**Comment by adangel** [comment, marked as answer]
Ideally, you wouldn't create the jar file manually. First of all, you use a build tool, such as maven. The template project for custom rules in java for java is here: https://github.com/pmd/pmd-examples/tree/java

However, you would run `mvn package` (which builds the jar file) only once you are finished with developing. While you are developing, you use a IDE like Eclipse or IntelliJ (anything that can import maven projects would do). There you can import your custom rule project and run the unit tests and debug etc. directly within the IDE, without building the jar file and copying it somewhere.
The IDE will make sure that your code changes are compiled.

**Reply by vedant011** [reply]
Can you guide to make a java rule class which detects the first block of catch and within a every catch statement there should be a first line having .error name otherwise it will generate a violation message. 
Ex. try {
  //  Block of code to try
}
catch(Exception e) {
this line must having logger.error(e)---------> specifically focused on .error only if it's not there then generate violation 
}
I'm stuck here due to confusion in pmd doc, i have created XPATH expression it's working fine but i want to create it through by second method of rule creation using java, please help me out i'm totally confused..

**Reply by adangel** [reply]
see https://github.com/pmd/pmd/discussions/4229

**Comment by vedant011** [comment]
Thank you..
